### PR TITLE
Make two-pass context phases explicit

### LINKED ISSUE #271: dynamic source contexts cannot distinguish phase 1 from a successful parser result of `undefined`
## Summary
The `SourceContext.getAnnotations(parsed?)` API uses `parsed === undefined` as the phase-1 sentinel, so if the first parse pass succeeds with an actual parser value of `undefined`, dynamic contexts cannot tell that phase 2 has started and behave as if they are still in phase 1.

## Reproduction
~~~~ typescript
import { createConfigContext, getActiveConfig } from "@optique/config";
import { runWith, runWithSync } from "@optique/core/facade";

const schema = {
  "~standard": {
    version: 1,
    vendor: "test",
    validate(value: unknown) {
      return { value: value as { host?: string } };
    },
  },
} as const;

const context = createConfigContext({ schema });

const parser = {
  $mode: "sync" as const,
  $valueType: [] as (string | undefined)[],
  $stateType: [] as null[],
  priority: 0,
  usage: [],
  initialState: null,
  parse(context: {
    readonly buffer: readonly string[];
    readonly state: null;
    readonly optionsTerminated: boolean;
    readonly usage: readonly unknown[];
  }) {
    return {
      success: true as const,
      next: context,
      consumed: [],
    };
  },
  complete() {
    return {
      success: true as const,
      value: getActiveConfig<{ readonly host?: string }>(context.id)?.host,
    };
  },
  *suggest() {},
  getDocFragments() {
    return { fragments: [] };
  },
};

console.log(await runWith(parser, "test", [context], {
  args: [],
  load: () => ({ config: { host: "config-host" }, meta: undefined }),
}));

console.log(runWithSync(parser, "test", [context], {
  args: [],
  load: () => ({ config: { host: "config-host" }, meta: undefined }),
}));
// both print undefined
~~~~

This is also saved locally as `tmp/repro-undefined-first-pass.ts`.

## Expected behavior
Dynamic contexts should receive an unambiguous phase-2 signal even when the first parse pass legitimately produces `undefined` as its value.

## Actual behavior
Both `runWith()` and `runWithSync()` behave as if phase 2 never happened, so the config loader is effectively skipped and the final value stays `undefined`.

## Notes
This is not limited to `createConfigContext()`. Any dynamic `SourceContext` implementation that follows the documented `parsed === undefined` phase-1 convention cannot support parsers whose successful result type includes `undefined` without some other out-of-band signal.

A real Optique parser shape can trigger this too. For example, `runWith(optional(flag("--debug")), "test", [context], { args: [], load: ... })` never calls the `load` callback when the parser succeeds with `undefined` on empty input.

### PR DESCRIPTION
**Pull request by dahlia**
## Summary

Fixes https://github.com/dahlia/optique/issues/271.

This PR removes the old sentinel-based two-pass `SourceContext` contract, where phase 1 was inferred from `parsed === undefined`, and replaces it with an explicit `SourceContextRequest` object. That ambiguity meant a successful first-pass value of `undefined` was observationally identical to “phase 1 has not happened yet”, so custom two-pass contexts could silently skip phase-two work. The runner now passes `{ phase: "phase1" }` and `{ phase: "phase2", parsed }` explicitly, which makes `undefined` a valid phase-two payload instead of a hidden control signal.

This is an intentional breaking API change for the 1.0.0 line. The core API changes live in *packages/core/src/context.ts* and *packages/core/src/facade.ts*. As part of that redesign, the old `finalizeParsed()` escape hatch was removed because it is no longer needed once the phase is explicit. The config integration in *packages/config/src/index.ts* was then simplified by deleting its private `undefined` wrapper workaround and switching directly to the new request model.

## Example

Custom two-pass contexts should now branch on `request.phase` instead of `parsed === undefined`:

```ts
const context: SourceContext = {
  id: myKey,
  phase: "two-pass",
  getAnnotations(request) {
    if (request == null || request.phase === "phase1") return {};
    return { [myKey]: loadFrom(request.parsed) };
  },
};
```

This keeps phase detection correct even when the first pass legitimately produces `undefined`.

## Tests and docs

The fix was developed TDD-first. Regression coverage was added in *packages/core/src/facade.test.ts* to prove that both `runWith()` and `runWithSync()` still enter phase 2 when the first-pass result is `undefined`. A config-focused regression was added in *packages/config/src/run.test.ts* to confirm that phase-two config loading still runs for top-level parsers whose successful value is `undefined`.

Manual `getAnnotations()` call sites and examples were updated to use the new request shape in *packages/config/src/index.test.ts*, *packages/env/src/index.test.ts*, *packages/inquirer/src/index.test.ts*, *packages/run/src/run.test.ts*, *examples/patterns/interactive-env-config-composition.ts*, and *examples/patterns/custom-source-context.ts*. The API docs and changelog were updated in *docs/concepts/extend.md*, *docs/concepts/runners.md*, *docs/cookbook.md*, *docs/integrations/config.md*, *docs/integrations/env.md*, and *CHANGES.md*.

## Validation

I verified this change with `deno task check`, `mise test`, and `pnpm build` in *docs/*.

### CODE REVIEW

File: packages/core/src/facade.test.ts [hunk 1]
```diff
@@ -51,6 +54,22 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
...
+  return request == null ||
+    (typeof request === "object" &&
+      "phase" in request &&
+      (request as { readonly phase?: unknown }).phase === "phase1");
+}
+
+function getPhase2ContextParsed<T>(request: unknown): T | undefined {
+  if (request != null && typeof request === "object" && "phase" in request) {
+    return (request as { readonly phase?: unknown }).phase === "phase2"
+      ? (request as SourceContextRequest & { readonly parsed: T }).parsed
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in 0c810e4ec7759e3e4cb46021304be34800e5bfef. The helper now narrows through SourceContextPhase2Request directly instead of the looser ad hoc cast.

File: packages/inquirer/src/index.test.ts [hunk 2]
```diff
@@ -1569,10 +1588,13 @@ describe("prompt()", () => {
         id: Symbol.for("@test/prompt-phase-two"),
         phase: "two-pass",
         getAnnotations(parsed?: unknown) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/inquirer/src/index.test.ts]: Addressed in 122db4268b19421ba79927ae50a815a22a986c1f. I renamed the callback parameter to request so the test matches the updated SourceContextRequest contract.

File: packages/core/src/facade.ts [hunk 3]
```diff
@@ -2935,6 +2923,10 @@ async function collectFinalAnnotations(
     deferred,
     deferredKeys,
   );
+  const request: SourceContextRequest = {
+    phase: "phase2",
+    parsed: preparedParsed,
+  };
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in 0c810e4ec7759e3e4cb46021304be34800e5bfef. Phase requests are now instantiated per context invocation, and I added async/sync regressions that cover cross-context request mutation.

File: packages/core/src/facade.test.ts [hunk 4]
```diff
@@ -51,6 +54,22 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
...
+      : undefined;
+  }
+  return request as T | undefined;
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts:78]: Addressed in b538dcfa1f21dd0c81b0f04c224c9061f41a227a. The phase-request helpers are now strict about explicit SourceContextRequest shapes, and phase-two checks now go through an explicit phase2 guard.

File: packages/core/src/facade.ts [hunk 5]
```diff
@@ -3048,26 +3039,20 @@ function collectFinalAnnotationsSync(
...
-        return internalAnnotations == null
-          ? result
-          : mergeAnnotations([result, internalAnnotations]);
-      },
+    const result = context.getAnnotations(request, options);
+    if (result instanceof Promise) {
+      throw new Error(
+        `Context ${String(context.id)} returned a Promise in sync mode. ` +
+          "Use runWith() or runWithAsync() for async contexts.",
+      );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:3047]: Addressed in b538dcfa1f21dd0c81b0f04c224c9061f41a227a. The sync/async contract violation now throws TypeError instead of a generic Error.

File: packages/inquirer/src/index.test.ts [hunk 6]
```diff
@@ -33,6 +36,22 @@ const promptFunctionsOverrideSymbol = Symbol.for(
...
+}
+
+function getPhase2ContextParsed<T>(request: unknown): T | undefined {
+  if (request != null && typeof request === "object" && "phase" in request) {
+    return (request as { readonly phase?: unknown }).phase === "phase2"
+      ? (request as SourceContextRequest & { readonly parsed: T }).parsed
+      : undefined;
+  }
+  return request as T | undefined;
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/inquirer/src/index.test.ts]: Addressed in b538dcfa1f21dd0c81b0f04c224c9061f41a227a. I tightened the inquirer test helpers so they only accept explicit SourceContextRequest shapes and no longer silently accept legacy raw payloads.

File: packages/run/src/run.test.ts [hunk 7]
```diff
@@ -29,6 +30,22 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
...
+}
+
+function getPhase2ContextParsed<T>(request: unknown): T | undefined {
+  if (request != null && typeof request === "object" && "phase" in request) {
+    return (request as { readonly phase?: unknown }).phase === "phase2"
+      ? (request as SourceContextRequest & { readonly parsed: T }).parsed
+      : undefined;
+  }
+  return request as T | undefined;
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/run/src/run.test.ts:53]: Addressed in b538dcfa1f21dd0c81b0f04c224c9061f41a227a. The run test helper now returns a parsed value only for explicit phase2 requests and otherwise returns undefined.

File: packages/core/src/facade.test.ts [hunk 8]
```diff
@@ -10212,6 +10238,55 @@ describe("branch coverage: facade.ts edge cases", () => {
...
+      readingContext,
+    ], {
+      args: [],
+    });
+
+    assert.deepEqual(secondContextParsed, { config: "optique.json" });
+    assert.deepEqual(result, {
+      config: "optique.json",
+    });
+  });
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts:10290]: I’m leaving this one as-is. The current branch at ca449c9055054e57cbb2421e794ebf7dece4d224 intentionally preserves parsed object identity when no scrub is needed, and that behavior is already locked in by the existing `preserves plain parsed object identity when no scrub is needed` regression. Mutating `parsed.config` here would start asserting a broader semantic change than this PR is making, whereas 0c810e4ec7759e3e4cb46021304be34800e5bfef specifically fixed the new wrapper-sharing issue introduced by the request object.

File: packages/core/src/facade.ts [hunk 9]
```diff
@@ -2983,15 +2969,16 @@ function collectPhase1AnnotationsSync(
...
   for (const context of contexts) {
-    const result = context.getAnnotations(undefined, options);
+    const request: SourceContextRequest = { phase: "phase1" };
+    const result = context.getAnnotations(request, options);
     if (result instanceof Promise) {
-      throw new Error(
+      throw new TypeError(
         `Context ${String(context.id)} returned a Promise in sync mode. ` +
           "Use runWith() or runWithAsync() for async contexts.",
       );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:2978]: Addressed in ca449c9055054e57cbb2421e794ebf7dece4d224. The public runWithSync() JSDoc now advertises TypeError for Promise-returning sync contexts and asyncDispose Promise violations, matching the actual runtime behavior.

File: docs/concepts/extend.md [hunk 10]
```diff
@@ -730,16 +730,25 @@ API reference
...
+    `getInternalAnnotations()`.
+
+    `phase: "phase1"`
+
+:   Initial annotation collection before the first parse pass.
+
+`phase: "phase2"`
+:   Second annotation collection after a usable first pass. The
+    `parsed` field holds the first-pass value, which may itself be
+    `undefined`.
```
[the remark this answers is not in the document]
Comment (dahlia) [reply docs/concepts/extend.md:750]: Addressed in ca449c9055054e57cbb2421e794ebf7dece4d224. I rewrote the SourceContextRequest entry to describe it explicitly as a discriminated union on the phase field and listed the two variants under that definition.

**Comment by dahlia** [comment]
@codex review

File: docs/concepts/extend.md [hunk 11]
```diff
@@ -891,10 +909,10 @@ const envContext: SourceContext = {
...
   phase: "two-pass",
-  async getAnnotations(parsed?: unknown) {
-    if (parsed === undefined) return {}; // Return empty on first pass
-    
-    const result = parsed as { config?: string };
+  async getAnnotations(request?: SourceContextRequest) {
+    if (request == null || request.phase === "phase1") return {};
+
+    const result = request.parsed as { config?: string };
     if (!result.config) return {};
```
[the remark this answers is not in the document]
Comment (dahlia) [reply docs/concepts/extend.md]: Addressed in e3a5d0cd75b0a1b52c36e4a08a43c74fa8923d40. The examples now treat request.parsed as possibly undefined in phase 2 and guard access via optional chaining before reading config.

File: packages/core/src/facade.test.ts [hunk 12]
```diff
@@ -5688,9 +5712,9 @@ describe("runWith", () => {
         id: configKey,
         phase: "two-pass",
         getAnnotations(parsed?: unknown) {
-          if (parsed === undefined) return {};
+          const result = getPhase2ContextParsed<{ config?: string }>(parsed);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in adb0b7bac2365af769671bb4a69490d3e2def533. I renamed that callback parameter to request to match the explicit SourceContextRequest contract.

File: packages/core/src/facade.test.ts [hunk 13]
```diff
@@ -11303,13 +11408,59 @@ describe("branch coverage: facade.ts edge cases", () => {
...
+        *suggest() {},
+        getDocFragments() {
+          return { fragments: [] };
+        },
+      };
+      const context: SourceContext = {
+        id: key,
+        phase: "two-pass",
+        getAnnotations(parsed?: unknown) {
+          if (isPhase1ContextRequest(parsed)) return {};
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in adb0b7bac2365af769671bb4a69490d3e2def533. I renamed the callback parameter to request here as well for consistency with the request-shaped two-pass contract.

File: packages/core/src/facade.test.ts [hunk 14]
```diff
@@ -11543,13 +11694,59 @@ describe("branch coverage: facade.ts edge cases", () => {
...
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in adb0b7bac2365af769671bb4a69490d3e2def533. This callback now uses request rather than parsed so the test reads consistently with the explicit phase request API.

File: packages/config/src/index.ts [hunk 15]
```diff
@@ -377,29 +370,27 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
       runtimeOptions?: unknown,
     ): Promise<Annotations> | Annotations {
       // Phase 1 (no parsed result): return no public annotations here.
       // Runners add the phase-1 unresolved marker through
       // getInternalAnnotations() so prompt(bindConfig(...)) can defer
       // interactive fallback without exposing that marker as user data.
-      if (parsed === undefined) {
+      if (request == null || request.phase === "phase1") {
         return {};
       }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:395]: Addressed in e3a5d0cd75b0a1b52c36e4a08a43c74fa8923d40. createConfigContext() now rejects malformed manual request objects instead of silently treating them as phase-two input, and there is a regression test covering the invalid request shape.

File: packages/core/src/facade.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:2978]: Addressed in adb0b7bac2365af769671bb4a69490d3e2def533. I updated the sync collector helper JSDoc so the @throws type now matches the TypeError thrown at runtime.

File: packages/core/src/facade.ts [hunk 16]
```diff
@@ -3048,26 +3035,24 @@ function collectFinalAnnotationsSync(
...
+    const request: SourceContextRequest = {
+      phase: "phase2",
+      parsed: preparedParsed,
+    };
+    const result = context.getAnnotations(request, options);
+    if (result instanceof Promise) {
+      throw new TypeError(
+        `Context ${String(context.id)} returned a Promise in sync mode. ` +
+          "Use runWith() or runWithAsync() for async contexts.",
+      );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:3047]: Addressed in adb0b7bac2365af769671bb4a69490d3e2def533. The @throws tag on collectFinalAnnotationsSync now documents TypeError instead of a generic Error.

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 17]
```diff
@@ -377,31 +370,41 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+      if (
+        typeof request !== "object" ||
+        !("phase" in request) ||
+        (request.phase !== "phase1" && request.phase !== "phase2")
+      ) {
+        throw new TypeError(
+          "Expected getAnnotations() to receive no request or a " +
+            'SourceContextRequest ({ phase: "phase1" } or ' +
+            '{ phase: "phase2", parsed }).',
+        );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:407]: Addressed in 185493c389aa2f8e0177291cf5b9b5aa11b21764. `createConfigContext().getAnnotations()` now rejects a `phase: "phase2"` request that omits `parsed`, and there is a regression test covering that malformed manual-call shape.

File: packages/core/src/facade.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:2978]: I’m leaving this one as-is. The current code already uses string concatenation across two string literals rather than a single long template literal, and the message already ends with a period. The interpolated value here is `context.id`, which is a symbol, not a locale-formatted number, so switching to `toLocaleString("en")` would not add meaningful precision.

File: packages/core/src/facade.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:3047]: Same rationale here: no code change needed. The error already ends with a period, the message is already split with string concatenation for readability, and `context.id` is a symbol so locale-specific numeric formatting is not relevant to this path.

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 18]
```diff
@@ -377,31 +370,42 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+        !("phase" in request) ||
+        (request.phase !== "phase1" && request.phase !== "phase2") ||
+        (request.phase === "phase2" && !("parsed" in request))
+      ) {
+        throw new TypeError(
+          "Expected getAnnotations() to receive no request or a " +
+            'SourceContextRequest ({ phase: "phase1" } or ' +
+            '{ phase: "phase2", parsed }).',
+        );
+      }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:408]: I’m leaving this one as-is. This validation is on a public API boundary (`createConfigContext().getAnnotations()` can be called manually), so a runtime shape check is intentional here rather than redundant internal-only validation. I agree that the guard-clause style would also work, but I’m not changing it in this PR because the current form is already explicit about the accepted request contract and keeps the error path localized.

File: packages/inquirer/src/index.test.ts [hunk 19]
```diff
@@ -33,6 +37,19 @@ const promptFunctionsOverrideSymbol = Symbol.for(
 
 let promptFunctionsOverrideQueue = Promise.resolve();
 
+function isPhase2ContextRequest(
+  request: unknown,
+): request is SourceContextPhase2Request {
+  return request != null &&
+    typeof request === "object" &&
+    "phase" in request &&
+    (request as { readonly phase?: unknown }).phase === "phase2";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/inquirer/src/index.test.ts]: Addressed in 7bef340eb4296a8a46e5c4ab79aaf83e54c8192c. The inquirer test helper now requires both `phase: "phase2"` and a `parsed` property before narrowing to SourceContextPhase2Request.

File: packages/core/src/facade.test.ts [hunk 20]
```diff
@@ -51,6 +55,26 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
...
+    (request as { readonly phase?: unknown }).phase === "phase1";
+}
+
+function isPhase2ContextRequest(
+  request: unknown,
+): request is SourceContextPhase2Request {
+  return request != null &&
+    typeof request === "object" &&
+    "phase" in request &&
+    (request as { readonly phase?: unknown }).phase === "phase2";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in 7bef340eb4296a8a46e5c4ab79aaf83e54c8192c. The core facade test helper now verifies that a phase-two request also carries `parsed`, so the type guard matches the request contract.

File: packages/run/src/run.test.ts [hunk 21]
```diff
@@ -29,6 +30,26 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
...
+    (request as { readonly phase?: unknown }).phase === "phase1";
+}
+
+function isPhase2ContextRequest(
+  request: unknown,
+): request is SourceContextPhase2Request {
+  return request != null &&
+    typeof request === "object" &&
+    "phase" in request &&
+    (request as { readonly phase?: unknown }).phase === "phase2";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/run/src/run.test.ts]: Addressed in 7bef340eb4296a8a46e5c4ab79aaf83e54c8192c. The run test helper now checks for both `phase: "phase2"` and the presence of `parsed` before narrowing.

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 22]
```diff
@@ -377,31 +370,42 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+      request?: SourceContextRequest,
       runtimeOptions?: unknown,
     ): Promise<Annotations> | Annotations {
       // Phase 1 (no parsed result): return no public annotations here.
       // Runners add the phase-1 unresolved marker through
       // getInternalAnnotations() so prompt(bindConfig(...)) can defer
       // interactive fallback without exposing that marker as user data.
-      if (parsed === undefined) {
+      if (request == null) {
         return {};
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts]: Addressed in cf72302221b489181775d775bc7df20af8a1925e. `createConfigContext().getAnnotations()` now treats only an omitted request as the manual phase-1 convenience path, while `null` is rejected through the same `TypeError` path as other malformed `SourceContextRequest` inputs. I also added a regression test covering `getAnnotations(null, options)`.

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 23]
```diff
@@ -377,31 +370,43 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+        !("phase" in request) ||
+        (request.phase !== "phase1" && request.phase !== "phase2") ||
+        (request.phase === "phase2" && !("parsed" in request))
+      ) {
+        throw new TypeError(
+          "Expected getAnnotations() to receive no request or a " +
+            'SourceContextRequest ({ phase: "phase1" } or ' +
+            '{ phase: "phase2", parsed }).',
+        );
+      }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:408]: Addressed in 2ab36573079ca7136e9eb601c10ce7c6b69695c3. `createConfigContext().getAnnotations()` now includes the received request type in its malformed-request `TypeError` messages, and the corresponding regression tests were updated to assert the more specific diagnostics. I left the validation inline rather than extracting a helper because this is a public API boundary and the current guard keeps the accepted request contract explicit.

File: packages/core/src/facade.test.ts [hunk 24]
```diff
@@ -5596,7 +5621,7 @@ describe("runWith", () => {
         id: Symbol.for("@test/phase-merge-late"),
         phase: "two-pass",
         getAnnotations(parsed?: unknown) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. I renamed the remaining `getAnnotations()` callback parameter from `parsed` to `request` in `packages/core/src/facade.test.ts` so these tests match the explicit `SourceContextRequest` contract.

File: packages/core/src/facade.test.ts [hunk 25]
```diff
@@ -5596,7 +5621,7 @@ describe("runWith", () => {
...
-          if (parsed == null) {
+          if (isPhase1ContextRequest(parsed)) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. The corresponding `isPhase1ContextRequest(...)` calls in the affected `facade.test.ts` cases now use the renamed `request` parameter as well.

File: packages/core/src/facade.test.ts [hunk 26]
```diff
@@ -5650,7 +5675,7 @@ describe("runWith", () => {
         id: Symbol.for("@test/phase-clear-early"),
         phase: "two-pass",
         getAnnotations(parsed?: unknown) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. I renamed this `getAnnotations()` parameter to `request` to keep the test terminology aligned with the new explicit phase-request API.

File: packages/core/src/facade.test.ts [hunk 27]
```diff
@@ -5650,7 +5675,7 @@ describe("runWith", () => {
...
-          if (parsed == null) {
+          if (isPhase1ContextRequest(parsed)) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. The associated `isPhase1ContextRequest(...)` check now uses `request` here too.

File: packages/core/src/facade.test.ts [hunk 28]
```diff
@@ -5661,7 +5686,7 @@ describe("runWith", () => {
         id: Symbol.for("@test/phase-clear-late"),
         phase: "two-pass",
         getAnnotations(parsed?: unknown) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. This remaining `parsed` parameter in `facade.test.ts` was renamed to `request` for consistency with `SourceContextRequest`.

File: packages/core/src/facade.test.ts [hunk 29]
```diff
@@ -5661,7 +5686,7 @@ describe("runWith", () => {
...
-          if (parsed == null) {
+          if (isPhase1ContextRequest(parsed)) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in bba1d28571e1c94bfe14f6606ac91109b671ef14. The corresponding reference here now uses the renamed `request` parameter.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.test.ts [hunk 30]
```diff
@@ -51,6 +55,27 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
 
 type AssertNever<T extends never> = T;
 
+function isPhase1ContextRequest(request: unknown): boolean {
+  return request != null &&
+    typeof request === "object" &&
+    "phase" in request &&
+    (request as { readonly phase?: unknown }).phase === "phase1";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Addressed in 37b2eca8a15c88d94719bf080c0eae1cb8368248. The phase-one test helper in `packages/core/src/facade.test.ts` now treats an omitted request (`undefined`) as the documented manual phase-one path, while still requiring an explicit request object for runner-provided phase checks. I intentionally did not widen it to `null`, since `null` is still treated as malformed input elsewhere in this PR.

File: packages/run/src/run.test.ts [hunk 31]
```diff
@@ -29,6 +30,27 @@ import { bindEnv, createEnvContext } from "../../env/src/index.ts";
 
 const TEST_DIR = join(import.meta.dirname ?? ".", "test-configs");
 
+function isPhase1ContextRequest(request: unknown): boolean {
+  return request != null &&
+    typeof request === "object" &&
+    "phase" in request &&
+    (request as { readonly phase?: unknown }).phase === "phase1";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/run/src/run.test.ts]: Addressed in 37b2eca8a15c88d94719bf080c0eae1cb8368248. The `packages/run/src/run.test.ts` helper now treats an omitted request (`undefined`) as the manual phase-one path documented by `SourceContext.getAnnotations()`, while keeping malformed non-request payloads out of the explicit phase-request checks.

File: packages/config/src/index.ts [hunk 32]
```diff
@@ -377,31 +370,43 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+        !("phase" in request) ||
+        (request.phase !== "phase1" && request.phase !== "phase2") ||
+        (request.phase === "phase2" && !("parsed" in request))
+      ) {
+        throw new TypeError(
+          "Expected getAnnotations() to receive no request or a " +
+            'SourceContextRequest ({ phase: "phase1" } or ' +
+            `{ phase: "phase2", parsed }), but got: ${getTypeName(request)}.`,
+        );
+      }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:408]: No code change here. The current implementation added in 2ab36573079ca7136e9eb601c10ce7c6b69695c3 is intentional: this is a public API boundary, so the request-shape validation stays explicit, the `"phase" in request` check is still required before reading `request.phase` from an unknown object, and `getTypeName()` is not a one-off helper in this file but the shared type-name formatter already used across the config input-validation paths.

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 33]
```diff
@@ -377,31 +370,43 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+        request === null ||
+        typeof request !== "object" ||
+        !("phase" in request) ||
+        (request.phase !== "phase1" && request.phase !== "phase2") ||
+        (request.phase === "phase2" && !("parsed" in request))
+      ) {
+        throw new TypeError(
+          "Expected getAnnotations() to receive no request or a " +
+            'SourceContextRequest ({ phase: "phase1" } or ' +
+            `{ phase: "phase2", parsed }), but got: ${getTypeName(request)}.`,
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:406]: No code change here. The current message introduced in 2ab36573079ca7136e9eb601c10ce7c6b69695c3 is already balanced as `SourceContextRequest ({ phase: "phase1" } or { phase: "phase2", parsed }), but got: ...`: the `}` closes the second request shape and the trailing `)` closes the outer parenthetical after both variants.

[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:406]: No code change here. `getTypeName()` is defined locally in the same file (see `packages/config/src/index.ts` above `createConfigContext()`), and the current malformed-request message path that uses it was added in 2ab36573079ca7136e9eb601c10ce7c6b69695c3. There is no missing import or runtime `ReferenceError` on this path.

File: packages/config/src/index.ts (hunk 32 above)
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts:408]: No code change here. The current implementation from 2ab36573079ca7136e9eb601c10ce7c6b69695c3 already includes the actual invalid type in the message via `getTypeName(request)`, and that helper intentionally reports `array` instead of the raw `typeof` result `object`. I’m also keeping the explicit shape check as-is because this is a public API boundary and the current form keeps the accepted `SourceContextRequest` contract localized and readable.
